### DISCUSSION TITLE
Next step ;) : small communities

### DISCUSSION THREAD

**Discussion by acarv** [simonbaird/tiddlyhost#193] [Ideas]
TiddlyHost is a fantastic tool!
The next step, IMHO, would be to allow small communities to form here, that is:
1) allow, for a site owner, to select other TiddlyHost users for a site community
2) allow, for a site owner, to grant edit privileges in the site for members of his community

This does not need much security or fine grained permissions (for now), just a way of building a small community around a site and collaboratively evolve it.
The TiddlyHost management part, however, should remain private to the single site owner.
The TiddlyHost management for each user should show the community sites he belongs to.

Just an improvement idea, TiddlyHost is fine as it is...

**Comment by simonbaird** [comment]
I like the idea, but I'm thinking we'll need a solid solution to prevent simultaneous updates from different users from overwriting each other.

**Reply by acarv** [reply]
Yes, I agree that concurrency is a challenge but maybe we can live with it for a while.
 [OokTech / TW5-Bob](https://github.com/OokTech/TW5-Bob) already implements multi-user under node, maybe its an useful example.
I don't know if TiddlyHost uses single file or file per tiddler approach internally but again we can have some kind of "lock" to notify other community members of the edit state.

Anyway, the idea is small communities so I would not mind to just have some kind of banner stating that a tiddler is being edited, its up to each user to respect that and the system may still work in "last writer wins". (I'm a CVS guy so do not feel the need for all the GIT complications ;) )
The Owner must be able to clear all those flags though and force a save.

On a very basic staged approach we would need:
1) a list of TiddlyHost users associated to a site
2) a write-allowed flag and a view-allowed flag (to cope with #164) for each user
3) at save time a check on that list to see if the user is on the write list

**Comment by simonbaird** [comment]
See also #164 which would have some overlap with the group/community management.
